### ISSUE TITLE
50 error patterns (Prior Art)

### ISSUE THREAD

**Issue by shaunlebron** [yogthos/clojure-error-message-catalog#7]
I wanted to mention that there exists a catalogue of 50 error patterns [documented with regex here](https://github.com/Clojure-Intro-Course/clojure-intro-class/blob/d1c291f3f72a663ec9342f26b6fa4a94b9bbc849/src/errors/error_dictionary.clj).  Here is the summary:

- assertion-error-with-argument
- assertion-error-without-argument
- class-cast-exception-cannot-cast-to-map-entry
- class-cast-exception
- assoc-parity-error
- wrong-number-of-args-passed-to-a-keyword
- illegal-argument-no-val-supplied-for-key
- illegal-argument-vector-arg-to-map-conj
- illegal-argument-cannot-convert-type
- illegal-argument-even-number-of-forms
- illegal-argument-even-number-of-forms-in-binding-vector
- illegal-argument-needs-vector-when-binding
- illegal-argument-type-not-supported
- illegal-argument-parameters-must-be-in-vector
- illegal-argument-exactly-2-forms
- duplicate-key-hashmap
- cant-call-nil
- compiler-exception-wrong-number-of-arguments-to-recur
- compiler-exception-on-improper-use-of-arg-literal
- index-out-of-bounds-index-provided
- string-index-out-of-bounds
- index-out-of-bounds-index-not-provided
- arity-exception-wrong-number-of-arguments
- null-pointer-non-existing-object-provided
- null-pointer-non-existing-object-not-provided
- unsupported-operation-wrong-type-of-argument
- compiler-exception-must-recur-from-tail-position
- class-not-found-exception
- number-format-exception
- lisp-reader-exception-long-error-error
- reader-tag-must-be-symbol
- invalid-tolken-error
- invalid-tolken-error
- syntax-error-cant-specifiy-over-20-args
- compiler-exception-first-argument-must-be-symbol
- compiler-exception-cannot-take-value-of-macro
- compiler-exception-cannot-resolve-symbol
- compiler-exception-map-literal-even
- compiler-exception-unmatched-delimiter
- compiler-exception-too-many-arguments
- compiler-exception-too-few-arguments
- compiler-exception-end-of-file
- compiler-exception-end-of-file-string
- compiler-exception-no-such-var-with-namespace
- compiler-exception-no-such-var-no-namespace
- compiler-exception-no-such-namespace
- java.lang.Exception-improper-identifier
- compiler-exception-even-numbers-in-binding-vector
- compiler-exception-even-number-of-forms-needed
- compiler-exception-wrong-number-of-arguments

**Comment by tolitius** [comment]
@shaunlebron that's great, might make sense to think about connecting this approach with errors we have / will have in this repo once we figure out the clean way for #6 

This catalog ultimately should be readable by programs as well as people before we can "run things on it"

Do you have an idea about the format (#6) or other ideas on how to incorporate this `error_dictionary` and alike?

**Comment by yogthos** [comment]
Yeah I agree that having a structured format at some point would be a good idea.

**Comment by shaunlebron** [comment]
@tolitius `error_dictionary` enumerates all the possible low-level error messages-- the message we would put in `Error Message` section in our error reports.

but given the direction of the current error reports mentioned in #14, there would be a many-to-one relationship between error reports and each exception listed by `error_dictionary`.  So I don't have a lot of clarity how to integrate this yet or if it even makes sense.

**Comment by eggsyntax** [comment]
The direction it seems to be going (see #16 ) is toward a single file per exception, so it seems pretty feasible to include the `error_dictionary`.

**Comment by yogthos** [comment]
Yeah, I agree as well.
